lapack64/lapack64.go

```diff
@@ -0,0 +1,42 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// lapack64 provides a set of convenient wrapper functions for LAPACK calls, as
+// specified in the netlib standard (www.netlib.org).
+//
+// The native Go routines are used by default, and the Use function can be used
+// to set an alternate implementation.
+//
+// If the type of matrix (General, Symmetric, etc.) is known and fixed, it is
+// used in the wrapper signature. In many cases, however, the type of the matrix
+// changes during the call to the routine, for example the matrix is symmetric on
+// entry and is triangular on exit. In these cases the correct types should be checked
+// in the documentation.
+//
+// The full set of Lapack functions is very large, and it is not clear that a
+// full implementation is desirable, let alone feasible. Please open up an issue
+// if there is a specific function you need and/or are willing to implement.
+package lapack64
+
+import (
+	"github.com/gonum/blas"
+	"github.com/gonum/lapack"
+	"github.com/gonum/lapack/native"
+)
+
+var lapack64 lapack.Float64 = native.Implementation{}
+
+// Use sets the LAPACK float64 implementation to be used by subsequent BLAS calls.
+// The default implementation is native.Implementation.
+func Use(l lapack.Float64) {
+	lapack64 = l
+}
+
+// Dpotrf computes the cholesky factorization of a.
+//  A = U^T * U if ul == blas.Upper
+//  A = L * L^T if ul == blas.Lower
+// On entry, a is a symmetric matrix, and at conclusion it is the U or L matrix.
+func Dpotrf(ul blas.Uplo, n int, a []float64, lda int) int {
```

Comment:
Yes, it should be Potrf.

I'm not sure about your suggestion for return types. Thinking about how I would want to use this in mat64
````
func (t *TriDense) Chol(a Symmetric) {
     // front matter
    // Copy data from SymDense into t
     lapack64.Potrf(t.mat.Uplo, t.mat.N, t.mat.Data, t.mat.Stride)
}
// And done because the data is now in-place into t
````
If we take a Symmetric in, we would need to copy the data into t, then construct a new blas64.Symmetric. This isn't costly, but I don't think it's any more clear for the code reader who needs to infer that the data is shared between the symmetric and the triangular.